### PR TITLE
[Tooling, Code Health] refactor: split Makefile into multiple files

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

The Makefile has gotten quite large ([1185 LoC](https://github.com/pokt-network/poktroll/blob/main/Makefile#L1185)). This PR splits the makefile into multiple files, focusing on large groups of targets and groups related to on-chain state or local/devnets:
- warnings.mk
- todos.mk
- checks.mk
- tests.mk
- localnet.mk
- testnet.mk
- params.mk
- applications.mk
- suppliers.mk
- gateways.mk
- session.mk
- claims.mk

## Issue

- Reaction to friction experienced while working on #799

## Type of change

Select one or more from the following:

- [ ] New feature, functionality or library
- [ ] Consensus breaking; add the `consensus-breaking` label if so. See #791 for details
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Documentation**: `make docusaurus_start`; only needed if you make doc changes
- [x] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: makefiles/checks.mk [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+.PHONY: fix_proto_unstable_marshalers
+fix_proto_unstable_marshalers: ## Ensure the `stable_marshaler_all` option is present on all protobuf files.
+	go run ./tools/scripts/protocheck/cmd unstable --fix
+	${MAKE} proto_regen
+
+
+.PHONY: warn_destructive
+warn_destructive: ## Print WARNING to the user
+	@echo "This is a destructive action that will affect docker resources outside the scope of this repo!"
```
Comment (bryanchriswhite) [inline makefiles/checks.mk:160]: move to warnings.mk

File: Makefile [hunk 2]
```diff
@@ -132,167 +132,6 @@ list: ## List all make targets
 help: ## Prints all the targets in all the Makefiles
```
Comment (Olshansk) [inline Makefile:132]: Did you confirm this still works?

File: makefiles/localnet.mk [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	@echo "Initializing chain..."
+	@set -e
+	@ignite chain init --skip-proto
+	AUTH_CONTENT=$$(cat ./tools/scripts/authz/dao_genesis_authorizations.json | jq -r tostring); \
+	$(SED) -i -E 's!^(\s*)"authorization": (\[\]|null)!\1"authorization": '$$AUTH_CONTENT'!' ${HOME}/.poktroll/config/genesis.json;
+
+	@cp -r ${HOME}/.poktroll/keyring-test $(POKTROLLD_HOME)
+	@cp -r ${HOME}/.poktroll/config $(POKTROLLD_HOME)/
+
+.PHONY: send_relay_sovereign_app_JSONRPC
```
Comment (Olshansk) [inline makefiles/localnet.mk:29]: Let's create a `relay.mk` as well

File: makefiles/localnet.mk [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	curl -X POST -H "Content-Type: application/json" \
+	--data '{"model": "qwen:0.5b", "stream": false, "messages": [{"role": "user", "content":"count from 1 to 10"}]}' \
+	$(APPGATE_SERVER)/ollama/api/chat
+
+.PHONY: cosmovisor_start_node
+cosmovisor_start_node: # Starts the node using cosmovisor that waits for an upgrade plan
+	bash tools/scripts/upgrades/cosmovisor-start-node.sh
+
+.PHONY: query_tx
+query_tx: ## Query for a transaction by hash and output as YAML (default).
```
Comment (Olshansk) [inline makefiles/localnet.mk:53]: Let's create a `query.mk` as well

**Review by Olshansk — APPROVED** [review]
This is great, ty!

**Review by Olshansk — APPROVED** [review]
This is great, ty!

**Review by Olshansk — APPROVED** [review]
This is great, ty!

File: Makefile (hunk 2 above)
Comment (bryanchriswhite) [reply Makefile:132]: Yessir! :saluting_face:

**Comment by bryanchriswhite** [comment]
Merging despite CI failures.

1. CI passed as of [335e68b](https://github.com/pokt-network/poktroll/pull/816/commits/335e68b6301eda037463e49c12ccf5e25a699559)
2. Subsequent commits only "merge with main", reverting commits that were merged from another branch into this branches original base (the PR base has since been updated to `main`).
3. This branch only modifies the makefiles
image

**Pull request merged**